src/test/java/org/cactoos/scalar/TernaryTest.java

```diff
@@ -73,14 +76,40 @@ void conditionBoolean() throws Exception {
 
     @Test
     void conditionFunc() throws Exception {
-        MatcherAssert.assertThat(
+        new Assertion<>(
+            "Must call the Func with the input",
             new Ternary<>(
                 5,
                 input -> input > 3,
-                input -> input = 8,
-                input -> input = 2
-            ).value(),
-            Matchers.equalTo(8)
-        );
+                input -> input + 1,
+                input -> input + 2
+            ),
+            new ScalarHasValue<>(6)
+        ).affirm();
+    }
+
+    @Test
+    void conditionSticky() throws Exception {
+        final AtomicInteger counter = new AtomicInteger(0);
+        new Assertion<>(
+            "Must conserve the same value for the whole evaluation",
+            new Ternary<String>(
+                counter::incrementAndGet,
+                i -> i == 1,
+                i ->  i + " equals " + 1,
+                i -> "else: " + i
```

Comment:
@victornoel It would be better as `String.format("else: %d")` here and in other pleases where `+` is used for concatenation.

Reply:
@andreoss I replaced all of this with `FormattedText`